tests/rbd/test_rbd_live_migrate_encryption.py

```diff
@@ -0,0 +1,539 @@
+    E.g: qemu-img convert -f raw -O qcow2 /dev/rbd0 /tmp/qimage1.qcow2
+    8.Create spec file with qcow format data
+    [root@ceph-sangadi-bz-sqd86e-node4 tmp]# cat /tmp/qcow_spec.json
+    {
+    "type": "qcow",
+    "stream": {
+    "type": "file",
+    "file_path": "/tmp/qimage1.qcow2"
+    }
+    }
+9. Make sure no client should holds that image before Initiate migration using
+    rbd migration prepare --import-only
+    E.g: rbd migration prepare --import-only --source-spec-path /tmp/qcow_spec.json m_pool/mr_image1
+    10. Execute and commit the migration
+    rbd migration execute pool/image
+    rbd migration commit pool/image
+11.  map that migrated image using rbd device map with encryption keys,
+    create mount directory and mount it and calculate md5sum and Cross-verify data consistency
+    E.g: rbd device map -t nbd -o encryption-format=luks2,encryption-passphrase-file=
+    /tmp/passphrase.bin m_pool/mr_image2
+12. Unmap the disk and cleanup the pools, namespace and image
+13. Repeat the test on LUKS2
+
+"""
+
+from copy import deepcopy
+
+from ceph.rbd.initial_config import initial_rbd_config
+from ceph.rbd.utils import (
+    check_data_integrity,
+    create_map_options,
+    getdict,
+    random_string,
+)
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.encryption import create_passphrase_file
+from ceph.rbd.workflows.krbd_io_handler import krbd_io_handler
+from ceph.rbd.workflows.migration import verify_migration_state
+from ceph.rbd.workflows.rbd import create_single_pool_and_images
+from ceph.utils import get_node_by_id
+from cli.rbd.rbd import Rbd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def migration_encrypted_raw_images(rbd_obj, client, **kw):
+    """
+        Test to verify Live migration of raw format images with encryption
+    Args:
+        rbd_obj: RBD object
+        client : client node object
+        **kw: any other arguments
+    """
+
+    kw["client"] = client
+    rbd = rbd_obj.get("rbd")
+
+    for pool_type in rbd_obj.get("pool_types"):
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = deepcopy(getdict(rbd_config))
+
+        for pool, pool_config in multi_pool_config.items():
+            kw["pool-name"] = pool
+            for encryption_type in kw.get("config", {}).get("encryption_type", {}):
+                kw.update({f"{pool}": {}})
+                kw[pool].update({"encryption_type": encryption_type})
+                kw[pool].update({"pool_type": pool_type})
+                image = (
+                    "image_" + kw[pool]["encryption_type"] + "_" + random_string(len=3)
+                )
+                kw[pool].update({"image": image})
+                err = run_io_on_encryption_formatted_image(rbd, pool, image, **kw)
+                if err:
+                    return 1
+
+                # Create spec file raw data
+                raw_file = "/tmp/" + image
```

Comment:
Can we use `tempfile` here instead of hardcoding

`raw_file = tempfile.mktemp(prefix=f"{image}_", suffix=".raw")`